
.. |pipeline_path| replace:: https://github.com/metagenlab/microbiome16S_pipeline/

.. _under_the_hood:
########################################################################
Under the hood
########################################################################

************************************************************************
Snakemake, environments and containers
************************************************************************

Snakemake
=======================================================================


Conda environments
=======================================================================


Singularity containers
=======================================================================


************************************************************************
Input output definition 
************************************************************************




Input
=======================================================================

:github:`A set of Python functions <rules/0_preprocessing/scripts/make_input_list.py>` is run at each execution of the RSP4ABM to match sequencing read files to each sample. 

As exposed in :ref:`the sample selection section of the Pipeline execution page <sample_selection>`, RSP4ABM can accept either local or SRA-hosted read files. 

Local reads
-----------------------------------------------------------------------

In the case of locally stored 


SRA reads
-----------------------------------------------------------------------

Output
=======================================================================


************************************************************************
Logging and traceability
************************************************************************

Snakemake logs
=======================================================================

RSP4ABM logs
=======================================================================



************************************************************************
Sequencing reads QC
************************************************************************



************************************************************************
Denoising
************************************************************************

Vsearch (OTU clustering)
=======================================================================

PANDAseq
-----------------------------------------------------------------------

Vsearch
-----------------------------------------------------------------------



DADA2 (ASV denoising)
=======================================================================

cutadapt
-----------------------------------------------------------------------

DADA2
-----------------------------------------------------------------------



************************************************************************
Taxonomic assignment
************************************************************************

reference database
=======================================================================

classifiers
=======================================================================



************************************************************************
Post-processing
************************************************************************


Taxonomic filtering
=======================================================================


Rarefaction
=======================================================================


Phylogenetic tree generation
=======================================================================


Taxonomic collapsing
=======================================================================


Normalization and abundance-based filtering
=======================================================================


Exports
=======================================================================


Fromatting
=======================================================================

Wide to long melting
-----------------------------------------------------------------------

transpose_and_meta_count_table
-----------------------------------------------------------------------

Qiime2 formats
-----------------------------------------------------------------------


************************************************************************
Picrust2
************************************************************************



